jsonschema-core.xml

```diff
@@ -800,49 +800,291 @@
+                    is being examined because it was identified as such by another
+                    schema's "$schema" keyword.  This means that a single schema
+                    document might sometimes be considered a regular schema, and
+                    other times be considered a meta-schema.
+                </t>
+                <t>
+                    In the case of examining a schema which is its own meta-schema,
+                    when an implementation begins processing it as a regular schema,
+                    it is processed under those rules.  However, when loaded a second
+                    time as a result of checking its own "$schema" value, it is treated
+                    as a meta-schema.  So the same document is processed both ways in
+                    the course of one session.
+                </t>
+                <t>
+                    Implementations MAY allow a schema to be passed as a meta-schema,
+                    for implementation-specific purposes, such as pre-loading a commonly
+                    used meta-schema and checking its vocabulary support requirements
+                    up front.  Meta-schema authors MUST NOT expect such features to be
+                    interoperable across implementations.
+                </t>
+            </section>
+            <section title="Best Practices for Vocabulary and Meta-Schema Authors">
+                <t>
+                    Meta-schema authors SHOULD NOT use "$vocabulary" to combine multiple
+                    vocabularies that define conflicting syntax or semantics for the same
+                    keyword.  As semantic conflicts are not generally detectable through
+                    schema validation, implementations are not expected to detect such
+                    conflicts.  If conflicting vocabularies are declared, the resulting
+                    behavior is undefined.
+                </t>
+                <t>
+                    Vocabulary authors SHOULD provide a meta-schema that validates the
+                    expected usage of the vocabulary's keywords.  Such meta-schemas
+                    SHOULD NOT forbid additional keywords, and MUST NOT forbid
+                    the "$id", "$schema", or "$vocabulary" keywords in the root schema.
+                </t>
+                <t>
+                    It is RECOMMENDED that meta-schema authors reference each vocabulary's
+                    meta-schema using the <xref target="allOf">"allOf"</xref> keyword,
+                    although other mechanisms for constructing the meta-schema may be
+                    appropriate for certain use cases.
+                </t>
+                <t>
+                    Meta-schemas MAY impose additional constraints, including describing
+                    keywords not present in any vocabulary, beyond what the meta-schemas
+                    associated with the declared vocabularies describe.  This allows for
```

Comment:
Yeah, that's actually a really good example.  You can describe those in a meta-schema using `patternProperties` (if you need to exclude them from a `false` `additionalProperties` or `unevaluatedProperties` which is how OAS works), but you can't declare them in a vocabulary because you don't know what keywords will actually be used in practice.  I might try to incorporate this concept in b/c it shows why this ability is needed.

Reply:
I ended up not adding this into the text because I fell down a philosophical rabbit-hole about whether the `x-` extension mechanism actually counts as being defined by the vocabulary specification (I think it does).  Let's just see what people try :-)

Reply:
I'm marking this resolved b/c I don't think you particularly needed me to add it, @Relequestual .  Feel free to re-open/re-comment if I missed something.